Doc/c-api/memory.rst

```diff
@@ -268,25 +268,34 @@ Customize Memory Allocators
-   +==========================================================+=======================================+
-   | ``void *ctx``                                            | user context passed as first argument |
-   +----------------------------------------------------------+---------------------------------------+
-   | ``void* malloc(void *ctx, size_t size)``                 | allocate a memory block               |
-   +----------------------------------------------------------+---------------------------------------+
-   | ``void* calloc(void *ctx, size_t nelem, size_t elsize)`` | allocate a memory block initialized   |
-   |                                                          | with zeros                            |
-   +----------------------------------------------------------+---------------------------------------+
-   | ``void* realloc(void *ctx, void *ptr, size_t new_size)`` | allocate or resize a memory block     |
-   +----------------------------------------------------------+---------------------------------------+
-   | ``void free(void *ctx, void *ptr)``                      | free a memory block                   |
-   +----------------------------------------------------------+---------------------------------------+
+   +-------------------------------------------------------------------+---------------------------------------+
+   | Field                                                             | Meaning                               |
+   +===================================================================+=======================================+
+   | ``void *ctx``                                                     | user context passed as first argument |
+   +-------------------------------------------------------------------+---------------------------------------+
+   | ``void* malloc(void *ctx, size_t size)``                          | allocate a memory block               |
+   +-------------------------------------------------------------------+---------------------------------------+
+   | ``void* calloc(void *ctx, size_t nelem, size_t elsize)``          | allocate a memory block initialized   |
+   |                                                                   | with zeros                            |
+   +-------------------------------------------------------------------+---------------------------------------+
+   | ``void* realloc(void *ctx, void *ptr, size_t new_size)``          | allocate or resize a memory block     |
+   +-------------------------------------------------------------------+---------------------------------------+
+   | ``void free(void *ctx, void *ptr)``                               | free a memory block                   |
+   +-------------------------------------------------------------------+---------------------------------------+
+   | ``void* aligned_alloc(void *ctx, size_t alignment, size_t size)`` | allocate an aligned memory block      |
```

Comment:
Do people have to provide `aligned_alloc` and `aligned_free`? Or can they leave those pointers NULL and get a default implementation?

Reply:
Currently, you must provide all allocator functions, included aligned_alloc and aligned_free.

Technically, we can implement a fallback, but I'm not sure that I want to do that :-)